frontend_tests/node_tests/voting_widget.js

```diff
@@ -76,4 +84,291 @@ run_test('poll_data_holder my question', () => {
+
+        poll_vote_button.attr('data-key', '100,1');
+        const e = {
+            stopPropagation: noop,
+            target: poll_vote_button,
+        };
+        out_data = undefined;
+        func(e);
+        assert.deepEqual(out_data, { type: 'vote', key: '100,1', vote: 1 });
+    };
+
+    widget_elem.handle_events(vote_events);
+
+    assert(poll_question.attr('disabled'));
+    assert(poll_question_input.attr('disabled'));
+
+    const add_ques_event = [
+        {
+            sender_id: 100,
+            data: {
+                type: 'question',
+                question: 'best plan?',
+            },
+        },
+    ];
+
+    poll_question_container.show();
+
+    widget_elem.handle_events(add_ques_event);
+
+    assert(!poll_question_container.visible());
+
+    {
+        const e = {
+            stopPropagation: noop,
+        };
+
+        // No event is send when it is not your poll
+        poll_question_input.val('Is it new?');
+        out_data = undefined;
+        poll_question_update_function(e);
+        assert.deepEqual(out_data, undefined);
+    }
+});
+
+run_test('activate own poll', () => {
+    $.clear_all_elements();
+
+    people.is_my_user_id = return_true;
+    templates.render = (template_name) => {
+        if (template_name === 'poll-widget') {
+            return 'poll-widget';
+        } else if (template_name === 'poll-widget-results') {
+            return 'poll-widget-results';
+        }
+    };
+
+    const widget_elem = $('<div>').addClass('widget-content');
+    let out_data;
+    const callback = (data) => {
+        out_data = data;
+    };
+    const opts = {
+        elem: widget_elem,
+        callback: callback,
+        message: {
+            sender_id: 100,
+        },
+        extra_data: {
+            question: 'Where to go?',
+        },
+    };
+
+    const poll_comment = $.create('button.poll-comment');
+    const poll_comment_input = $.create('input.poll-comment');
+    const widget_comment_container = $.create('ul.poll-widget');
+
+    const poll_question = $.create('button.poll-question');
+    const poll_question_input = $.create('input.poll-question');
+    const poll_question_header = $.create('.poll-question-header');
+    const poll_question_container = $.create('.poll-question-bar');
+
+    const poll_vote_button = $.create('button.poll-vote');
+
+    widget_elem.set_find_results('button.poll-comment', poll_comment);
+    widget_elem.set_find_results('input.poll-comment', poll_comment_input);
+    widget_elem.set_find_results('button.poll-question', poll_question);
+    widget_elem.set_find_results('input.poll-question', poll_question_input);
+    widget_elem.set_find_results('.poll-question-header', poll_question_header);
+    widget_elem.set_find_results('ul.poll-widget', widget_comment_container);
+    widget_elem.set_find_results('button.poll-vote', poll_vote_button);
+    widget_elem.set_find_results('.poll-question-bar', poll_question_container);
```

Comment:
ohh, I should extract this into a function?

Reply:
Not sure how much of this you intended to put in a function, but I'm not sure scoping should get in your way here.